lib/src/query.dart

```diff
@@ -0,0 +1,766 @@
+import "bindings/helpers.dart";
+import "bindings/structs.dart";
+import "bindings/signatures.dart";
+import "package:ffi/ffi.dart";
+
+// TODO a future or stream(?) should be attachable to the Condition types
+// TODO in case there is an error, or some internal message that should be exposed
+// TODO do async
+/**
+ * The QueryProperty types are responsible
+ * for the operator overloading.
+ *
+ * A QueryBuilder will be constructed,
+ * based on the any / all operations applied.
+ *
+ * When build() is called on the QueryBuilder
+ * a Query object will be created.
+ */
+class QueryProperty {
+  int propertyId; // why hide this if it's exposed in the entity definition, // TODO final / const?
+  int entityId;
+  QueryProperty(this.entityId, this.propertyId);
+
+  QueryCondition isNull() {
+    // the bool serves as a dummy type, to initialize the base type
+    final c = Condition<bool>(ConditionOp._null, null, false);
+    return new QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition notNull() {
+    final c = Condition<int>(ConditionOp._not_null, null, 0);
+    return new QueryCondition(entityId, propertyId, c);
+  }
+}
+
+class QueryStringProperty extends QueryProperty {
+  QueryStringProperty(int entityId, int propertyId) : super(entityId, propertyId);
+
+  static const ConditionType type = ConditionType._string;
+
+  QueryCondition _op(String p, ConditionOp cop, bool caseSensitive, bool descending) {
+    final c = StringCondition(cop, type, p, null, caseSensitive ? OBXOrderFlag.CASE_SENSITIVE : 0, descending ? OBXOrderFlag.DESCENDING : 0);
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition _opWithEqual(String p, ConditionOp cop, bool caseSensitive, bool withEqual) {
+    final c = StringCondition._withEqual(cop, type, p, caseSensitive ? OBXOrderFlag.CASE_SENSITIVE : 0, withEqual);
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition _opList(List<String> list, ConditionOp cop, bool caseSensitive) {
+    final c = StringCondition._fromList(cop, type, list, caseSensitive ? OBXOrderFlag.CASE_SENSITIVE : 0);
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition equal(String p, {bool caseSensitive = false}) {
+    return _op(p, ConditionOp._eq, caseSensitive, false);
+  }
+
+  QueryCondition notEqual(String p, {bool caseSensitive = false}) {
+    return _op(p, ConditionOp._not_eq, caseSensitive, false);
+  }
+
+  QueryCondition endsWith(String p, {bool descending = false}) {
+    return _op(p, ConditionOp._string_ends, false, descending);
+  }
+
+  QueryCondition startsWith(String p, {bool descending = false}) {
+    return _op(p, ConditionOp._string_starts, false, descending);
+  }
+
+  QueryCondition contains(String p, {bool caseSensitive = false}) {
+    return _op(p, ConditionOp._string_contains, caseSensitive, false);
+  }
+
+  // TODO rename to something better
+  QueryCondition contain(String p, {bool caseSensitive = false}) {
```

Comment:
I was wondering what it does.

Reply:
It is supported in the C-API, it's `PropertyType::StringVector`

basically - it if you store a vector of strings in a property tagged this way, it can run some queries (I think right now it's only "contains"). It's a useful thing and should definitely be covered later (as in in the "all-type-support issue") in Dart as well

Reply:
Done